src/ophyd_async/core/_detector.py

```diff
@@ -21,7 +21,7 @@
     Triggerable,
     WritesStreamAssets,
 )
-from event_model import DataKey
+from event_model import DataKey  # type: ignore
```

Comment:
Do we need the `# type: ignore` here?

Reply:
Does `pyright` also complain? If it needs a `py.typed` then please could you make an event-model PR to add it?

Reply:
pyright does not seem to complain, event-model is missing a py.typed marker so I can add it

Reply:
PR here: https://github.com/bluesky/event-model/pull/346

Reply:
@coretl It looks like adding this `py.typed` file to event-model breaks bluesky's tests since there are a lot of unresolved typing issues. I can work on these typing issues but I don't think this should be a blocker for this PR to be merged. Let me know if you think otherwise.